helpers/utils.py

```diff
@@ -0,0 +1,111 @@
+    SPARK_STATIC_DEFAULTS_FILE = f"{os.environ.get('SNAP')}/conf/spark-defaults.conf"
+    return SPARK_STATIC_DEFAULTS_FILE
+
+def get_dynamic_defaults_conf_file() -> str:
+    SPARK_DYNAMIC_DEFAULTS_FILE = f"{os.environ.get('SNAP_USER_DATA')}/spark-defaults.conf"
+    return SPARK_DYNAMIC_DEFAULTS_FILE
+
+def get_env_defaults_conf_file() -> str:
+    SPARK_CONF_DIR = os.environ.get('SPARK_CONF_DIR', f"{os.environ.get('SPARK_HOME')}/conf")
+    SPARK_ENV_DEFAULTS_FILE = os.environ.get('SNAP_SPARK_ENV_CONF', f"{SPARK_CONF_DIR}/spark-defaults.conf")
+    return SPARK_ENV_DEFAULTS_FILE
+
+def get_snap_temp_dir() -> str:
+    return '/tmp/snap.spark-client'
+
+def parse_conf_overrides(conf_args: List) -> Dict:
+    conf_overrides = dict()
+    if conf_args:
+        for c in conf_args:
+            try:
+                kv = c.split('=')
+                k = kv[0]
+                v = '='.join(kv[1:])
+                conf_overrides[k] = os.environ.get(v, v)
+            except IndexError as e:
+                logging.error('Configuration related arguments parsing error. Please check input arguments and try again.')
+                sys.exit(EXIT_CODE_BAD_CONF_ARG)
+    return conf_overrides
+
+def reconstruct_submit_args(args: List, conf: Dict) -> List:
+    submit_args = args
+    conf_arg = ''
+    for k in conf.keys():
+        conf_arg += f' --conf {k}={conf[k].strip()}'
+    submit_args = [conf_arg] + submit_args
+    return submit_args
+
+def autodetect_kubernetes_master(conf: Dict) -> str:
+    USER_HOME_DIR = pwd.getpwuid(os.getuid())[USER_HOME_DIR_ENT_IDX]
+    DEFAULT_KUBECONFIG = f'{USER_HOME_DIR}/.kube/config'
+    kubeconfig = os.environ.get('KUBECONFIG') or DEFAULT_KUBECONFIG
+    kubectl_cmd = '{}/kubectl'.format(os.environ['SNAP'])
+    context = conf.get('spark.kubernetes.context')
+    if context:
+        master_kubectl_cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig} --context {context} config view --minify -o jsonpath=\"{{.clusters[0]['cluster.server']}}\""
+    else:
+        master_kubectl_cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig} config view --minify -o jsonpath=\"{{.clusters[0]['cluster.server']}}\""
+    try:
+       default_master = subprocess.check_output(master_kubectl_cmd, shell=True)
+       k8s_master_uri = 'k8s://' + default_master.decode("utf-8")
+    except subprocess.CalledProcessError as e:
+       logging.error(f'Cannot determine default kubernetes control plane url. Probably bad or missing kubeconfig file {kubeconfig}')
+       logging.warning(f'Please set the KUBECONFIG environment variable to point to the kubeconfig. Or fix the default kubeconfig {kubeconfig}')
+       logging.error(e.output)
+       sys.exit(EXIT_CODE_BAD_KUBECONFIG)
+    return k8s_master_uri
+
+def UmaskNamedTemporaryFile(*args, **kargs):
+    fdesc = NamedTemporaryFile(*args, **kargs)
+    umask = os.umask(0o666)
+    os.umask(umask)
+    os.chmod(fdesc.name, 0o666 & ~umask)
+    return fdesc
```

Comment:
Please can we add some unit tests for the script and some functional tests that demonstrate everything works, thanks